### PR TITLE
Remove krb5_get_max_time_skew portability hack

### PR DESCRIPTION
**Pull request by yadij**
This function has been available in Kerberos libraries
for over ten years now.

### CODE REVIEW

**Comment by yadij** [comment]
`krb5_get_max_time_skew(3)` has been part of libkrb5 from MIT and Heimdal for over 10 years now.
If this change turns out to be controversial I will just drop the portability support for old Heimdal
libraries entirely.

File: src/peer_proxy_negotiate_auth.cc [hunk 1]
```diff
@@ -307,13 +306,7 @@ int krb5_create_cache(char *kf, char *pn) {
                    error_message(code));
             return (1);
         }
-#elif HAVE_LIBHEIMDAL_KRB5 && HAVE_KRB5_GET_MAX_TIME_SKEW
         skew = krb5_get_max_time_skew(kparam.context);
```
Comment (rousskov) [inline src/peer_proxy_negotiate_auth.cc:309]: In my question below, the phrase "profile condition" stands for `HAVE_PROFILE_H && HAVE_KRB5_GET_PROFILE && HAVE_PROFILE_GET_INTEGER && HAVE_PROFILE_RELEASE` condition a few lines above this code. I also assume, like this PR does, that we have access to krb5_get_max_time_skew().

In official code, this line is compiled only when profile condition is false.

In PR code, this line is compiled only when profile condition is true.

If this change is not intentional, please fix.

File: src/peer_proxy_negotiate_auth.cc [hunk 2]
```diff
@@ -307,13 +306,7 @@ int krb5_create_cache(char *kf, char *pn) {
...
-#elif HAVE_LIBHEIMDAL_KRB5 && HAVE_MAX_SKEW_IN_KRB5_CONTEXT
-        skew = kparam.context->max_skew;
-#else
-        skew = DEFAULT_SKEW;
-#endif
```
Comment (rousskov) [inline src/peer_proxy_negotiate_auth.cc:316]: This PR removes one `#endif`. It does not remove any `#if`s. That feels like a PR bug and an indication that primary PR changes were not tested. Github [CI](https://github.com/squid-cache/squid/actions/runs/10186412841/job/28178141740?pr=1875) appears to agree with me:

```
peer_proxy_negotiate_auth.cc:15: error: unterminated #if
```

File: acinclude/krb5.m4 [hunk 3]
```diff
@@ -122,6 +122,28 @@ krb5_context kc; kc->max_skew = 1;
...
 
+dnl ensure krb5_get_max_time_skew(3) is defined for old Heimdal libraries
+AC_DEFUN([SQUID_CHECK_KRB5_GET_MAX_TIME_SKEW],[
+  AC_CACHE_CHECK([how to access max_skew in struct krb5_context],squid_cv_krb5_get_max_time_skew,[
+    SQUID_STATE_SAVE(squid_krb5_test)
+    CPPFLAGS="-I${srcdir:-.} $CPPFLAGS"
+    AC_CHECK_LIB([krb5],[krb5_get_max_time_skew],
+      [squid_cv_krb5_get_max_time_skew="function"],[
+      AC_COMPILE_IFELSE(
+        [AC_LANG_PROGRAM([[#include "compat/krb5.h"]],[[krb5_context kc; kc->max_skew = 1;]])],
```
[the remark this answers is not in the document]
Comment (rousskov) [reply acinclude/krb5.m4]: The above change request was written before PR branch was force-pushed (and the associated code changes removed from PR branch). Now the opposite question comes to mind: Should this PR remove SQUID_CHECK_MAX_SKEW_IN_KRB5_CONTEXT definition (as unused)?

**Review by rousskov — CHANGES_REQUESTED** [review]
I am not going to judge whether krb5_get_max_time_skew() is available in enough environments we care about. Your call.

File: acinclude/krb5.m4 (hunk 3 above)
[the remark this answers is not in the document]
Comment (yadij) [reply acinclude/krb5.m4]: Ah, yes. Thanks.

File: src/peer_proxy_negotiate_auth.cc (hunk 1 above)
Comment (yadij) [reply src/peer_proxy_negotiate_auth.cc:311]: fixed in 7cce5b64e5

File: src/peer_proxy_negotiate_auth.cc (hunk 2 above)
Comment (yadij) [reply src/peer_proxy_negotiate_auth.cc:316]: fixed in 7cce5b64e5

**Review by rousskov — COMMENTED** [review]
> This function has been available in Kerberos libraries for over ten years now.

Can we say the same about any other functions/features checked in SQUID_CHECK_KRB5_FUNCS and such? It is not clear why we are "picking on" krb5_get_max_time_skew() and/or whether we should be removing similar Kerberos hacks as well.

**Comment by yadij** [comment]
We can possibly say the same about other things. This one was adding complications to the other refactoring and was quick enough to verify.

If there are no actual issues I would like to merge this in the next two days and continue with the larger work.

**Review by kinkie — APPROVED** [review]
let's ship it

**Comment by kinkie** [comment]
@yadij approving, but stopping short of shipping

**Comment by yadij** [comment]
Looks like we had resolved this a while back but not cleared it. Doing that now.

**Pull request closed without merging**